File: Java/Program/Product.java
package Java.Program; 

class Product extends Clothing{ // Kelas Product yang merupakan subkelas dari kelas Clothing

    private String idProduct; // Deklarasi variabel private string untuk id produk
    private String name; // Deklarasi variabel private string untuk nama
    private String brand; // Deklarasi variabel private string untuk brand
    private int price; // Deklarasi variabel private integer untuk harga

    void Product() { // Deklarasi konstruktor untuk kelas Product
        idProduct = ""; // Melakukan inisialisai id produk 
        name = ""; // Melakukan inisialisai nama 
        brand = ""; // Melakukan inisialisai brand 
        price = 0; // Melakukan inisialisai harga
    }

    void set_idProduct(String idProduct) { // Deklarasi metode untuk mengatur id produk
        this.idProduct = idProduct; // Menetapkan id produk dengan nilai yang diberikan
    }

    void set_name(String name) { // Deklarasi metode untuk mengatur nama
        this.name = name; // Menetapkan nama dengan nilai yang diberikan
    }

    void set_brand(String brand) { // Deklarasi metode untuk mengatur brand
        this.brand = brand; // Menetapkan brand dengan nilai yang diberikan
    }

    void set_price(int price) { // Deklarasi metode untuk mengatur harga
        this.price = price; // Menetapkan harga dengan nilai yang diberikan
    }

    String get_idProduct() { // Deklarasi metode untuk mendapatkan id produk
        return this.idProduct; // Mengembalikan id produk
    }

    String get_name() { // Deklarasi metode untuk mendapatkan nama
        return this.name; // Mengembalikan nama
    }

    String get_brand() { // Deklarasi metode untuk mendapatkan brand
        return this.brand; // Mengembalikan brand
    }

    int get_price() { // Deklarasi metode untuk mendapatkan harga
        return this.price; // Mengembalikan harga
    }
}

File: Python/Program /Product.py
# Mengimpor kelas Clothing
from Clothing import Clothing

# Mendefinisikan kelas Product yang merupakan turunan dari kelas Clothing
class Product(Clothing):
    # Konstruktor untuk kelas Product
    def __init__(self):
        # Memanggil konstruktor dari kelas induk (Clothing)
        super().__init__()
        # Menambahkan atribut idProduct dengan nilai awal string kosong
        self.idProduct = ""
        # Menambahkan atribut name dengan nilai awal string kosong
        self.name = ""
        # Menambahkan atribut brand dengan nilai awal string kosong
        self.brand = ""
        # Menambahkan atribut price dengan nilai awal 0
        self.price = 0

    # Metode untuk mengatur nilai dari atribut idProduct
    def set_idProduct(self, idProduct):
        self.idProduct = idProduct

    # Metode untuk mengatur nilai dari atribut name
    def set_name(self, name):
        self.name = name

    # Metode untuk mengatur nilai dari atribut brand
    def set_brand(self, brand):
        self.brand = brand

    # Metode untuk mengatur nilai dari atribut price
    def set_price(self, price):
        self.price = price

    # Metode untuk mendapatkan nilai dari atribut idProduct
    def get_idProduct(self):
        return self.idProduct

    # Metode untuk mendapatkan nilai dari atribut name
    def get_name(self):
        return self.name

    # Metode untuk mendapatkan nilai dari atribut brand
    def get_brand(self):
        return self.brand

    # Metode untuk mendapatkan nilai dari atribut price
    def get_price(self):
        return self.price
